scipy/special/tests/test_stirling.py

```diff
@@ -43,13 +45,34 @@
     (6, 6, 1),
     (6, 7, 0),
 ])
-def test_triangle(n, k, expected):
-    assert stirling2(n, k) == expected
+def test_triangle_exact(n, k, expected):
+    assert stirling2(n, k, exact=True) == expected
 
 
-def test_modulus():
-    # do not need to fix seed
-    n = randrange(7, 12)
-    k = randrange(n)
+@pytest.mark.parametrize('n,expected', [
+    (0, [1]),
+    (1, [0, 1]),
+    (2, [0, 1, 1]),
+    (3, [0, 1, 3, 1]),
+    (4, [0, 1, 7, 6, 1]),
+    (5, [0, 1, 15, 25, 10, 1]),
+    (6, [0, 1, 31, 90, 65, 15, 1]),
+    (7, [0, 1, 63, 301, 350, 140, 21, 1]),
+    (8, [0, 1, 127, 966, 1701, 1050, 266, 28, 1]),
+    (9, [0, 1, 255, 3025, 7770, 6951, 2646, 462, 36, 1]),
+    (10, [0, 1, 511, 9330, 34105, 42525, 22827, 5880, 750, 45, 1]),
+])
+def test_triangle_inexact(n, expected):
+    """Test full triangle of values for 0 <= n <= 10
```

Comment:
```suggestion
    """Test triangle of values for 0 <= n <= 10 and 0 <= k <= n
```

not to be pedantic but "full" could mean `-inf < k < inf` to some people

Reply:
Not pedantic. You're right